### DISCUSSION TITLE
Section 03: My doubt is if model_0 is able to process 32 images all together

### DISCUSSION THREAD

**Discussion by lejan** [mrdbourke/pytorch-deep-learning#838] [Q&A]
Chapter 103: Writing training and testing loop for our batched data (minute 20)

```
# Import tqdm for progress bar
from tqdm.auto import tqdm

# Set the seed and start the timer
torch.manual_seed(42)
train_time_start_on_cpu = timer()

# Set the number of epochs (we'll keep this small for faster training times)
epochs = 20

# Create training and testing loop
for epoch in tqdm(range(epochs)):
    print(f"Epoch: {epoch}\n-------")
    ### Training
    train_loss = 0
    # Add a loop to loop through training batches
    for batch, (X, y) in enumerate(train_dataloader):
        model_0.train() ....
```

My doubt is: is model_0 processing the whole batch in one run?

**Comment by IlanVinograd** [comment]
Yes, when you use model_0 (or any PyTorch model) in a loop that iterates through a DataLoader, the model processes each batch of data in one run during each iteration of the loop.

In PyTorch, a DataLoader is typically set up to load batches of data. The batch_size parameter in the DataLoader constructor specifies how many samples from your dataset to load in one batch. When you iterate over the DataLoader, each iteration yields a batch of data (X, y), where X is a tensor representing the batch of input features, and y is a tensor representing the batch of labels.
